packages/xstate-graph/src/graph.ts

```diff
@@ -171,182 +167,141 @@ export function getAdjacencyMap<
-      const [weight] = weightMap.get(vertex)!;
-      for (const event of Object.keys(adjacency[vertex])) {
-        const nextSegment = adjacency[vertex][event];
-        const nextVertex = optionsWithDefaults.stateSerializer(
-          nextSegment.state
-        );
-        stateMap.set(nextVertex, nextSegment.state);
-        if (!weightMap.has(nextVertex)) {
-          weightMap.set(nextVertex, [weight + 1, vertex, event]);
+    for (const serializedState of unvisited) {
+      const [weight] = weightMap.get(serializedState)!;
+      for (const event of Object.keys(
+        adjacency[serializedState].transitions
+      ) as SerializedEvent[]) {
+        const { state: nextState, event: eventObject } = adjacency[
+          serializedState
+        ].transitions[event];
+        const nextSerializedState = serializeState(nextState, eventObject);
+        stateMap.set(nextSerializedState, nextState);
+        if (!weightMap.has(nextSerializedState)) {
+          weightMap.set(nextSerializedState, [
+            weight + 1,
+            serializedState,
+            eventObject
+          ]);
         } else {
-          const [nextWeight] = weightMap.get(nextVertex)!;
+          const [nextWeight] = weightMap.get(nextSerializedState)!;
           if (nextWeight > weight + 1) {
-            weightMap.set(nextVertex, [weight + 1, vertex, event]);
+            weightMap.set(nextSerializedState, [
+              weight + 1,
+              serializedState,
+              eventObject
+            ]);
           }
         }
-        if (!visited.has(nextVertex)) {
-          unvisited.add(nextVertex);
+        if (!visited.has(nextSerializedState)) {
+          unvisited.add(nextSerializedState);
         }
       }
-      visited.add(vertex);
-      unvisited.delete(vertex);
+      visited.add(serializedState);
+      unvisited.delete(serializedState);
     }
   }
 
-  const statePathMap: StatePathsMap<TContext, TEvent> = {};
+  const statePlanMap: StatePlanMap<TState, TEvent> = {};
 
   weightMap.forEach(([weight, fromState, fromEvent], stateSerial) => {
     const state = stateMap.get(stateSerial)!;
-    statePathMap[stateSerial] = {
+    statePlanMap[stateSerial] = {
       state,
       paths: !fromState
         ? [
             {
               state,
-              segments: [],
+              steps: [],
               weight
             }
           ]
         : [
             {
               state,
-              segments: statePathMap[fromState].paths[0].segments.concat({
+              steps: statePlanMap[fromState].paths[0].steps.concat({
                 state: stateMap.get(fromState)!,
-                event: deserializeEventString(fromEvent!) as TEvent
+                event: fromEvent!
               }),
               weight
             }
           ]
     };
   });
 
-  return statePathMap;
+  return Object.values(statePlanMap);
 }
 
-export function getSimplePaths<
+export function getSimplePlans<
```

Comment:
what's the current purpose of this helper? what's a meaningful difference between this and the `traverseSimplePlans` (that is being called here)?

Reply:
is the main difference that `get*` functions accept a state machine whereas the `traverse*` ones receive a behavior?

Reply:
Ye, that difference is not obvious at all. It will be somewhat hard to name them differently unless we literally put machine/behavior in their respective names. So perhaps we should reach out for different entrypoints? or export machine-oriented helpers from the root entry and use `/behaviors` for those other ones?

In general `get*` clearly indicates that we are going to get something in return, while with `traverse*` I was somewhat expecting a provided callback to be called on each state/path/something (that's what often happens in AST-related projects like Babel and TypeScript)

Reply:
Yeah, the naming can be improved. Will revise

Reply:
Done